usr_management.c

```diff
@@ -0,0 +1,216 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+#define USER_FILE "users.txt"
+#define MAX_LEN 100
+
+typedef struct {
+    int id;
+    char name[MAX_LEN];
+    int age;
+} User;
+
+
+int is_duplicate(int id) {
+    FILE *file = fopen(USER_FILE, "r");
+    if (!file) {
+        perror("Error opening file");
+        return 0; 
+    }
+
+    User user;
+    while (fscanf(file, "%d,%[^,],%d\n", &user.id, user.name, &user.age) == 3) {
+        if (user.id == id) {
+            fclose(file);
+            return 1; 
+        }               
+    }
+
+    fclose(file);
+    return 0; 
+}
+
+void init_file() {
+    FILE *file = fopen(USER_FILE, "r");
+    if (!file) {
+       
+        file = fopen(USER_FILE, "w");
+        if (!file) {
+            perror("Error creating file");
+            exit(1);
+        }
+        printf("File '%s' created successfully.\n", USER_FILE);
+    } else {
+        printf("File '%s' already exists.\n", USER_FILE);
+    }
+    fclose(file);
+}
+
+void add_user() {
+    FILE *file = fopen(USER_FILE, "a+");

```

Comment:
this  piece of code is repeating in other functions as well. Can you make a separate function and use it everywhere.

Reply:
Sorry sir,  but can you please specify what part of the code exactly. Is it the file opening statement ? or the user details input in the add_user function?